feat: Add reprocessing docs (#2907)
getsentry/sentry-docs: src/docs/product/error-monitoring/reprocessing/index.mdx

  enabled. **If the original minidump no longer exists in Sentry, sending an
  event through reprocessing will cause it to have no stack trace at all.**

* **Quota applies.** Every event you choose to reprocess will count against
  your plan's quota a second time. Rate limits and spike protection do not
  apply.

* **Debug files don't apply immediately.** When uploading debug information
  files, regardless of whether you upload to your own symbol server or to Sentry, the debug information files are
  applied only one hour after internal caches have expired. Please wait
  at least an hour after uploading DIFs before reprocessing.

* **Alerts do not apply, with caveats.** Issue alerts do not trigger for
  reprocessed events, and the new events are not subject to data forwarding.

  Metric alerts generally do not trigger either; however, they are subject to the
  aforementioned data inconsistencies. If you have created a metric alert with
  a 24-hour measurement window, and an event you reprocess happened in the last
  24 hours, it will temporarily be counted twice. This may trigger a metric
  alert.

> ## Legacy Reprocesing

Author: ```suggestion
## Legacy Reprocessing
```

---
title: Reprocessing
sidebar_order: 10

> description: Learn about reprocessing errors with new debug files and sourcemaps.

Reviewer: ```suggestion
description: Learn about reprocessing errors with new debug files and source maps.
```

---

> After you have uploaded new [debug information
> files](/platforms/native/data-management/debug-files/) or
> [sourcemaps](/platforms/javascript/sourcemaps/), Sentry only uses them for new
> events going forward. _Reprocessing_ allows you to apply them to older events as
> well.

Reviewer: ```suggestion
Sentry uses [debug information
files](/platforms/native/data-management/debug-files/) or
[source maps](/platforms/javascript/sourcemaps/) only after they have been uploaded. _Reprocessing_ allows you to apply source maps and debug information files to error events that have already occurred.
```

<Note>

> This feature is available only if you're in the Early Adopter program. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony.

Author: I found this note in workflow-related docs and copypasted it here

If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.

</Note>

After becoming an Early Adopter, you will find a new button when viewing an issue:

!["Reprocess this issue" button next to Resolve, Ignore, Share, Delete actions in issue details page](reprocessing-btn.png)

> The dialogue behind it asks you two questions:

Reviewer: after some internal debate (and noting that dialog is not in our Style Guide, so will be added):
```suggestion
The dialog requires two answers:
```

> * _How many events to reprocess_ -- When left empty, all events are reprocessed
>   and potentially move to other (new or existing) issues if the stacktrace
>   changed significantly. When 20 is entered, the 20 most recent events will be
>   reprocessed and the rest of the issue will be left intact.

Reviewer: ```suggestion
* _How many events to reprocess_ -- When left empty, all events are reprocessed
  and potentially move to other (new or existing) issues if the stack trace
  changed significantly. If you provide a number, for example, 20, then the 20 most recent events are
  reprocessed, and any remaining issue are left intact.
```

> * _What to do with remaining events_ -- If you choose to reprocess only 20
>   events out of 100, you can choose what to do with the remaining 80 events.
>   The default is to keep them where they are (the issue you're viewing).
>   Alternatively you may choose to delete them.

Reviewer: ```suggestion
* _What to do with remaining events_ -- If you choose to reprocess, for example, only 20
  events out of 100, you can choose what happens to the remaining 80 events.
  The default is to keep them where they are (the issue you're viewing).
  Alternately, you may choose to delete them.
```

> After hitting _Reprocess Events_, **the issue will be locked and hidden from

Reviewer: ```suggestion
After pressing _Reprocess Events_, **the issue will be locked and hidden from
```

your issue stream while reprocessing is ongoing**. As data is migrated between

> issues we allow no actions such as resolving or assigning the issue to ensure

Reviewer: ```suggestion
issues we allow no actions, such as resolving or assigning the issue, to ensure
```

data consistency and avoid data loss.

Note that you can still get to the issue manually by searching for
`is:reprocessing` (removing all other search terms) to get around those
restrictions if you have to.

## Caveats

> The dialogue for starting reprocessing lists a couple of caveats that you
> should really familiarize yourself with and understand before starting.

Reviewer: ```suggestion
The dialog to start reprocessing notes a few caveats that you
should be aware of and understand prior to kicking off reprocessing.
```

* **Data glitches.** During reprocessing you may observe temporary data
  inconsistencies across the entire product. You may observe an event belong to
  two issues at once, and appear in Discover queries twice.

>   Those inconsistencies disappear the second reprocessing is done, however
>   depending on your platform and the amount of events involved this can take a
>   while.

Reviewer: ```suggestion
  Those inconsistencies disappear the moment reprocessing is complete. However,
  depending on your platform and the amount of events involved, reprocessing may take a
  while.
```

> * **Attachment storage needs to be enabled.** If your events come from
>   minidumps or unreal crash reports, you need to make sure to have [attachment
>   storage](/platforms/native/enriching-events/attachments/#crash-reports-and-privacy)
>   enabled. **If the original minidump no longer exists in Sentry, sending an
>   event through reprocessing will cause it to have no stacktrace at all.**

Reviewer: ```suggestion
* **Attachment storage needs to be enabled.** If your events come from
  minidumps or unreal crash reports, you must have [attachment
  storage](/platforms/native/enriching-events/attachments/#crash-reports-and-privacy)
  enabled. **If the original minidump no longer exists in Sentry, sending an
  event through reprocessing will cause it to have no stack trace at all.**
```

* **Quota applies.** Every event you choose to reprocess will count against
  your plan's quota a second time. Rate limits and spike protection do not
  apply.

> * **Debug files don't apply immediately.** When uploading debug information
>   files, regardless whether to your own symbol server or to Sentry, they are
>   only applied after one hour after internal caches have expired. Please wait
>   at least 1 hour after uploading DIFs before reprocessing.

Reviewer: ```suggestion
* **Debug files don't apply immediately.** When uploading debug information
  files, regardless of whether you upload to your own symbol server or to Sentry, the debug information files are
  applied only one hour after internal caches have expired. Please wait
  at least an hour after uploading DIFs before reprocessing.
```

* **Alerts do not apply, with caveats.** Issue alerts do not trigger for
  reprocessed events, and the new events are not subject to data forwarding.

  Metric alerts generally do not trigger either, however are subject to the
  aforementioned data inconsistencies. If you have created a metric alert with
  a 24-hour measurement window, and an event you reprocess happened in the last

>   24 hours, it will temporarily be counted twice. This may set off the metric

Reviewer 2: what does temporarily mean? while the issue being reprocessed?

Author: yup. the double-count should disappear after reprocessing

>   Metric alerts generally do not trigger either, however are subject to the
>   aforementioned data inconsistencies. If you have created a metric alert with
>   a 24-hour measurement window, and an event you reprocess happened in the last
>   24 hours, it will temporarily be counted twice. This may set off the metric
>   alert.

Reviewer: ```suggestion
  Metric alerts generally do not trigger either; however, they are subject to the
  aforementioned data inconsistencies. If you have created a metric alert with
  a 24-hour measurement window, and an event you reprocess happened in the last
  24 hours, it will temporarily be counted twice. This may trigger a metric
  alert.
```

> ## Legacy Reprocesing

Reviewer 2: typo in reprocessing

Sentry can suspend incoming crash reports until all required debug information
files have been uploaded. **This feature is also called Reprocessing.** It can
be configured in _Project Settings > Processing Issues_. By default, this
feature is disabled.

If enabled, crash reports with missing debug files will not be displayed in the
issues stream. Instead, you will receive a warning that events cannot be
processed until all debug files have been uploaded.

> It is not recommended to enable this project setting as it has several flaws,
> the most important of which is that it does not work properly with minidumps or
> Unreal crash reports. Reprocessing existing issues as shown above is supposed
> to replace this functionality.

Reviewer: ```suggestion
We do not recommend enabling this project setting as it does not work properly with minidumps or
Unreal crash reports. Reprocessing existing issues as shown above replaces this functionality.
```
